lib/build.js

```diff
@@ -124,7 +124,59 @@ exports.filterIgnoreAndFiles = function(dir, ignore, files) {
           return;
         throw e;
       });
-    }));
+    }))
+    // remove empty folders
+    .then(function() {
+
+      //strip base dir from allFiles first to correctly compare it with the removeFiles
+      for (var i = allFiles.length - 1; i >= 0; i--) {
+        if(allFiles[i].indexOf(dir+'/' === 0)) allFiles[i] = allFiles[i].substr(dir.length+1);
+      }
+
+      // create object listing directories by file counts from the remaining files
+      var dirFileCount = {};
+      //var dirFileRemainCount = {};
+      allFiles.forEach(function(file) {
+        var dir = file.substr(0, file.lastIndexOf('/'));
+        dirFileCount[dir] = dirFileCount[dir] + 1 || 1;
+      });
+
+      // decrement file counts from removeFiles
+      removeFiles.forEach(function(file) {
+        var dir = file.substr(0, file.lastIndexOf('/'));
+        dirFileCount[dir] = dirFileCount[dir] - 1;
+      });
+
+      // now go through and populate removeDirs
+      var removeDirs = [];
+      Object.keys(dirFileCount).forEach(function(rdir) {
+        if (rdir!=='' && dirFileCount[rdir] === 0)
+          removeDirs.push(dir+'/'+rdir);
+      });
+
+      //add files from the removeFiles array that look like directories (do not have file extension)
+      //this is needed to pickup empty directories inside empty directories
+      removeFiles.forEach(function(file) {
+        var dotpos = file.lastIndexOf('.');
```

Comment:
This check isn't enough because files don't need to have extensions.

Rather I was suggesting explicitly detecting `EISDIR` in https://github.com/jspm/jspm-cli/pull/567/files#diff-26b681bd4dac10f25d5293c7595c435bR123 and then running an rmdir up there which would avoid the issue.

Sorry to be so pedantic about this, but it needs to be comprehensive. Happy to polish it myself to when I can get to it.